### PR TITLE
refactor: move upgrade unit tests together

### PR DESCRIPTION
**Pull request by ninabarbakadze**
## Overview

Moves all upgrade unit tests in one place

### CODE REVIEW

**Review by rootulp — APPROVED** [review]
Thanks!

File: test/util/testapp_upgrade.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package util
```
Comment (rootulp) [inline test/util/testapp_upgrade.go:1]: [not blocking] FYI: https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common

File: test/util/testapp_upgrade.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/celestiaorg/celestia-app/v2/app"
+	"github.com/celestiaorg/celestia-app/v2/app/encoding"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func SetupTestApp(t *testing.T, upgradeHeight int64) (*app.App, keyring.Keyring) {
```
Comment (cmwaters) [inline test/util/testapp_upgrade.go:17]: nit: 
```suggestion
func SetupTestAppWithUpgradeHeight(t *testing.T, upgradeHeight int64) (*app.App, keyring.Keyring) {
```

**Review by cmwaters — APPROVED** [review]
The other approach would be to say that these are all upgrade tests and just to bundle them up together rather than adding a new test for each new module

**Review by rootulp — APPROVED** [review]

**Review by cmwaters — CHANGES_REQUESTED** [review]
You're still repeating yourself a lot here. This kind of pattern lends itself to table based testing.

```go
testCases := []struct {
	module string
	preUpgradeAssertion func (t *testing.T, app *app.App)
	postUpgradeAssertion func(t *testing.T, app *app.App)
}{
	{
		module: "signal",
		preUpgradeAssertion: func(t *testing.T, app *app.App, ) {
			// add assertions about how this module doesn't yet exist
		},
		postUpgradeAssertion: func(t *testing.T, app *app.App) {
			// add assertions about how this module now exists
		},
	},
}

// start up the testnet

// do the pre upgrade assertions for all testCases

// upgrade the application and check it succeeded

// do all the post upgrade assertions for all testCases

```

Also don't forget that when a test errors that it tells the caller which module failed and at which stage

**Review by cmwaters — APPROVED** [review]
Looks good once you fix the conflicts

**Review by cmwaters — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**